### ISSUE TITLE
subresource "hosting environment" has space in it and is valid

### ISSUE THREAD

**Issue by vasu-telepathy** [hashicorp/terraform-provider-azurerm#25309]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.5

### AzureRM Provider Version

3.96.0

### Affected Resource(s)/Data Source(s)

azurerm_private_endpoint

### Terraform Configuration Files

```hcl
azurerm_private_endpoint does not take spaces in subresource. However, "hosting environment" is a valid subresource type and needs to be supported.
```

### Debug Output/Panic Output

```shell
╷
│ Error: private_service_connection.0.subresource_names.0 must begin and end with a alphanumeric character, be between 3 and 63 characters in length, only contain letters, numbers, underscores, periods, and dashes
│ 
│   with azurerm_private_endpoint.example,
│   on lambda.tf line 91, in resource "azurerm_private_endpoint" "example":
│   91:     subresource_names              = ["hosting environment","sites"]
│ 
╵
Releasing state lock. This may take a few moments...
Error: Process completed with exit code 1.
```

### Expected Behaviour

We should support this special subresource type as per this documentation: https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-overview#private-link-resource

### Actual Behaviour

Throws an error and validation fails for any spaces in the subresource string.

**Comment by vasu-telepathy** [comment]
@wuxu92 Yep. Tried that. That time I got a validation error or something of not being a valid sub resource.

**Comment by vasu-telepathy** [comment]
@wuxu92 Yes. It is V3. Yeah. But, looking at the azurem code-base, our validation throws an error if we have spaces. 
I think in the validation if it is not this above subresource then and then only we should throw an error for spaces.

**Comment by pier62350** [comment]
Have the same problem here with "Power BI" for Microsoft.PowerBI/privateLinkServicesForPowerBI private endpoint..

`Error: private_service_connection.0.subresource_names.0 must begin and end with a alphanumeric character, be between 3 and 63 characters in length, only contain letters, numbers, underscores, periods, and dashes
│ 
│   with module.common.azurerm_private_endpoint.this,
│   on main.tf line 49, in resource "azurerm_private_endpoint" "this":
│   49:     subresource_names              = ["Power BI"]`

**Comment by algattik** [comment]
Reproduces on azurerm 4.81.0 with another affected subresource: Azure Data Manager for Energy (ADME), whose private-link target sub-resource is literally `Azure Data Manager for Energy` (spaces included). This value is [documented](https://learn.microsoft.com/en-us/azure/energy-data-services/how-to-set-up-private-links) by Microsoft.
